src/evo/providertx.cpp

```diff
@@ -108,7 +109,25 @@ bool CheckFutureTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValid
     return true;
 }
 
-bool CheckNewAssetTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValidationState& state)
+
+inline bool checkNewUniqueAsset(CNewAssetTx& assettx, CValidationState& state)
+{
+    if (assettx.decimalPoint > 0 ){ // alway 0
+        return state.DoS(100, false, REJECT_INVALID, "bad-assets-decimalPoint"); 
+    }
+    
+    if (assettx.type > 0 ){ // manual mint only?
+        return state.DoS(100, false, REJECT_INVALID, "bad-unique-assets-distibution-type");
+    }
+
+    if (assettx.Amount != 1 * COIN){ //unique suply = 1 COIN
```

Comment:
an unique asset does not mean supply is always 1. unique asset mean each individual asset coin tie to an identity so each asset coin may not created equals. these indentation (id) that associated with each coin is specified in asset output script. for example, let say a unique asset token name Taylor_Swift_concert_ticket (TSCT for short), it first mint transaction produce 100 TSCT, the mint transaction would have asset amount = 100 and the output script would contain id=[1-100]. we want to use range here as much as possible to avoid listing 1,2,3...., 100. now user transfer 1 TSCT to another address, they can specify which id they sending. and the change output would have the remaining id and amounts. In this example TSCT[1] would represent a front row seat at the concert and would be more example then TSCT[100] which let it is back row. 

This change may be a lot so i suggest we approve this PR with other changes request above and make another PR for this unique asset.